Rakefile

```diff
@@ -71,6 +72,81 @@ task :encrypt_secrets do
   Encrypted::Environment.encrypt_ejson("secrets.ejson", private_key: ENV["SECRET_KEY"])
 end
 
+desc("Benchmarks tuist against a specified versiom")
+task :benchmark do
+
+  print_section("🛠 Building supporting tools")
+
+  # Build tuistbench
+  system(
+    "swift", "build", 
+    "-c", "release", 
+    "--package-path", "tools/tuistbench"
+  )
+
+  # Build fixturegen
+  system(
+    "swift", "build", 
+    "-c", "release", 
+    "--package-path", "tools/fixturegen"
+  )
+
+  # Generate large fixture
+  print_section("📁 Generating fixtures ...")
+  FileUtils.mkdir_p("generated_fixtures")
+
+  system(
+    "tools/fixturegen/.build/release/fixturegen",
+    "--path", "generated_fixtures/50_projects",
+    "--projects", "50",
+  )
+
+  system(
+    "tools/fixturegen/.build/release/fixturegen",
+    "--path", "generated_fixtures/2000_sources",
+    "--projects", "2",
+    "--sources", "2000",
+  )
+
+  # Generate fixture list
+  fixtures = {
+    "paths" => [
+      "generated_fixtures/50_projects",
+      "generated_fixtures/2000_sources",
+      "fixtures/ios_app_with_static_frameworks",
+      "fixtures/ios_app_with_framework_and_resources",
+      "fixtures/ios_app_with_transitive_framework",
+      "fixtures/ios_app_with_xcframeworks",
+    ]
+  }
+  File.open(".fixtures.generated.json","w") do |f|
+    f.write(fixtures.to_json)
+  end
+
+  # Build current version of tuist
+  print_section("🔨 Building release version of tuist ...")
+  system("swift", "build", "--product", "tuist", "--configuration", "release")
+  system("swift", "build", "--product", "tuistenv", "--configuration", "release")
+  system("swift", "build", "--product", "ProjectDescription", "--configuration", "release")
+
+  # Download latest tuist
+  print_section("⬇️ Downloading latest published version of tuist ...")
+
+  system(".build/release/tuistenv", "update")
+  puts("Reference tuist version:")
+  system(".build/release/tuistenv", "version")
+
+  print_section("⏱ Benchmarking ...")
+  system(
```

Comment:
@andreacipriani what about doing a similar implementation for benchmarking the time it takes to generate the the hashes of the graph nodes? I think we could have something like:

```
namespace :benchmark do
  task :generate do
    # Code
  end

  task :hashes do
    # Code
  end
end
```

Thoughts @kwridan ?

Reply:
Ah that would be great!

`tuistbench` was designed to allow benchmarking any of the tuist commands :) - by default it uses `generate` but we can use any other parameter too :)

e.g.

https://github.com/tuist/tuist/tree/master/tools/tuistbench#usage

```json
{
    "arguments": ["hash"],
    "runs": 5,
    "deltaThreshold": 0.02
}
```